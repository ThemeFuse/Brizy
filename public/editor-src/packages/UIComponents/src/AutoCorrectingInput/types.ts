import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
