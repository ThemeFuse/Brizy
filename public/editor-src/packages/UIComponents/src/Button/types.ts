import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
