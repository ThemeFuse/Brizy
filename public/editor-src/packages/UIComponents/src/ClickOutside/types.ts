import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
