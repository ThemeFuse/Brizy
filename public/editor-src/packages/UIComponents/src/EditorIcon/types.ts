export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
