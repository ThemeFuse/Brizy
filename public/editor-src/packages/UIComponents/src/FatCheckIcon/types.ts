import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
