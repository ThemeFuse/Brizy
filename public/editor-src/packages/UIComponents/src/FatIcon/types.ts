import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
