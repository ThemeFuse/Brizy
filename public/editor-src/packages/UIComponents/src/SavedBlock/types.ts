import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  title?: string;
  tooltipContent?: string;
  isLoading: boolean;
  icon: IconsName;
}
