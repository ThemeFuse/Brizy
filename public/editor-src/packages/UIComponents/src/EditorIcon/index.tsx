import Loadable from "@loadable/component";
import classnames from "classnames";
import React, { CSSProperties, FC, MouseEvent, ReactElement } from "react";
import { IconsName } from "./types";

const LoadableIcon = Loadable(async () => {
  const { Icon } = await import("./components/Icon");
  return Icon;
});
export interface Props {
  className?: string;
  icon: IconsName;
  style?: CSSProperties;
  onClick?: (e: MouseEvent<SVGElement>) => void;
}

export const EditorIcon: FC<Props> = ({
  className,
  icon,
  style,
  onClick
}): ReactElement => {
  const _className = classnames(
    "brz-icon-svg brz-ed-icon-svg align-[initial]",
    className
  );

  return (
    <LoadableIcon
      name={icon}
      className={_className}
      style={style}
      onClick={onClick}
    />
  );
};
