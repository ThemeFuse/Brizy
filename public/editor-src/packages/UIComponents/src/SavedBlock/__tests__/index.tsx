import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { IconsName } from "../../EditorIcon/types";
import { SavedBlock } from "../index";

describe("Test SavedBlock", () => {
  const onClick = jest.fn();

  describe("Snapshots", () => {
    test("Render loader", async () => {
      const savedBlock = render(
        <SavedBlock
          icon={IconsName.Heart}
          isLoading
          onClick={onClick}
          title="Save"
          tooltipContent="Saved"
        />
      );

      await waitFor(() => {
        expect(savedBlock.container.firstChild).toBeInTheDocument();
      });

      const loadingIcon = savedBlock.container.querySelector(
        ".brz-ed-animated--spin"
      );
      expect(loadingIcon).toBeInTheDocument();
      expect(savedBlock.container.firstChild).toMatchSnapshot();
    });

    test("Render Icon", async () => {
      const savedBlock = render(
        <SavedBlock
          icon={IconsName.Heart}
          isLoading={false}
          onClick={onClick}
          title="Save"
          tooltipContent="Saved"
        />
      );

      await waitFor(() => {
        expect(savedBlock.container.firstChild).toBeInTheDocument();
      });

      const icon = savedBlock.container.querySelector(
        ".brz-ed-animated--fadeIn"
      );
      expect(icon).toBeInTheDocument();
      expect(savedBlock.container.firstChild).toMatchSnapshot();
    });

    test("Check title prop", () => {
      render(
        <SavedBlock
          icon={IconsName.Heart}
          isLoading={false}
          onClick={onClick}
          title="Save"
          tooltipContent="Saved"
        />
      );
      const titleElement = screen.getByTitle("Save");
      expect(titleElement).toBeInTheDocument();
    });

    test("Check tooltip render", () => {
      const { container } = render(
        <SavedBlock
          icon={IconsName.Heart}
          isLoading={false}
          onClick={onClick}
          title="Save"
          tooltipContent="Saved"
        />
      );

      const tooltip = container.querySelector(".brz-ed-toolbar__item__tooltip");
      expect(tooltip).toBeInTheDocument();
    });

    test("If loading is true dont render tooltip", () => {
      const { container } = render(
        <SavedBlock
          icon={IconsName.Heart}
          isLoading
          onClick={onClick}
          title="Save"
          tooltipContent="Saved"
        />
      );

      const tooltip = container.querySelector(".brz-ed-toolbar__item__tooltip");
      expect(tooltip).not.toBeInTheDocument();
    });

    test("If no tooltip provided dont render tooltip", () => {
      const { container } = render(
        <SavedBlock
          icon={IconsName.Heart}
          isLoading={false}
          onClick={onClick}
          title="Save"
        />
      );

      const tooltip = container.querySelector(".brz-ed-toolbar__item__tooltip");
      expect(tooltip).not.toBeInTheDocument();
    });
  });

  test("Check click event", () => {
    const { container } = render(
      <SavedBlock
        icon={IconsName.Heart}
        isLoading={false}
        onClick={onClick}
        title="Save"
        tooltipContent="Saved"
      />
    );
    const element = container.firstElementChild;

    element && fireEvent.click(element);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
