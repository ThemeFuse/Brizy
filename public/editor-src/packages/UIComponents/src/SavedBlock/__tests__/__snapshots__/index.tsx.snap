// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SavedBlock Snapshots Render Icon 1`] = `
<div
  class="brz-ed-tooltip__content"
  title="Save"
>
  <div
    class="brz-ed-toolbar__item__tooltip"
  >
    Saved
  </div>
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial] brz-ed-animated brz-ed-animated--fadeIn"
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g
      fill="currentColor"
      fill-rule="nonzero"
      transform="translate(-86 -419)"
    >
      <g
        transform="translate(47)"
      >
        <g
          transform="translate(0 78)"
        >
          <g
            transform="translate(0 328)"
          >
            <g
              transform="translate(39 11)"
            >
              <g
                transform="translate(0 1)"
              >
                <path
                  d="M14.682 2.318a4.5 4.5 0 00-6.364 0c-.121.121-.214.259-.318.389-.104-.13-.197-.268-.318-.389a4.5 4.5 0 00-6.364 6.364L8 15l6.682-6.318a4.5 4.5 0 000-6.364z"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Test SavedBlock Snapshots Render loader 1`] = `
<div
  class="brz-ed-tooltip__content"
  title="Save"
>
  <svg
    class="brz-icon-svg brz-ed-icon-svg align-[initial] brz-ed-animated--spin"
    fill="currentColor"
    height="1em"
    viewBox="0 0 16 16"
    width="1em"
  >
    <g>
      <g>
        <g>
          <path
            class="cls-1"
            d="M8,16a8,8,0,1,1,8-8A8,8,0,0,1,8,16ZM8,2a6,6,0,1,0,6,6A6,6,0,0,0,8,2Z"
            fill="currentColor"
            style="opacity: 0.4;"
          />
          <path
            d="M16,8H14A6,6,0,0,0,8,2V0A8,8,0,0,1,16,8Z"
            fill="currentColor"
          />
        </g>
      </g>
    </g>
  </svg>
</div>
`;
